plugins/node/instrumentation-mongoose/src/utils.ts

```diff
@@ -99,15 +99,27 @@ export function handleCallbackResponse(
   originalThis: any,
   span: Span,
   responseHook?: MongooseResponseCustomAttributesFunction,
-  moduleVersion: string | undefined = undefined
+  moduleVersion: string | undefined = undefined,
+  ...args: IArguments[]
 ) {
-  return exec.apply(originalThis, [
-    (err: Error, response: any) => {
-      err
-        ? setErrorStatus(span, err)
-        : applyResponseHook(span, response, responseHook, moduleVersion);
-      span.end();
-      return callback!(err, response);
-    },
-  ]);
+  const newArgs = [];
+  for (const [, argValue] of Object.entries(args[0])) {
```

Comment:
The argument is an array but we only consume the first value which we expect to be an object. Can you explain the motivation for this?

Is `args[0]` the options object? in case why do we need to desturct it's values into an array? and what if it is being used with both options and a callback?

Reply:
Hi @blumamir , thanks for taking the time to review!

Sorry, this was my mistake. I, for some reason, defined `args` as `...IArguments[]` and assumed it was necessary.
So, just for explaining the code:
Logging the the value of `args` when the method is called with both the `options` object and the `callback` function, we would get the following:
```
args: [ [Arguments] { '0': [Object], '1': [AsyncFunction (anonymous)] } ],
```
In this case, `args` would be an array that has only 1 element - the array of the method arguments: `options` and `callback`. Here is what loging the value of `args[0]` would result:
```
args0: [Arguments] {
    '0': { session: [ClientSession] },
    '1': [AsyncFunction (anonymous)]
  }
```
-----
But, yes, this is not necessary. 
I am pushing an update where we can make the update `args` is defined as `IArguments` and we can directly update `args`.